components/linking.ts

```diff
@@ -0,0 +1,516 @@
+import {NavigationState, PartialState} from '@react-navigation/native';
+import {Logger} from 'browser-bunyan';
+import {TabNavigatorParamList} from 'routes';
+import {AvalancheCenterID, AvalancheCenterWebsites, reverseLookup} from 'types/nationalAvalancheCenter';
+
+export const getStateFromUrl = (logger: Logger, rawUrl: string): undefined | PartialState<NavigationState<TabNavigatorParamList>> => {
```

Comment:
TODO: on unknown, fall back to a new webview in the root stack that just shows the website, since there's no way to programatically reject some parts of the domain if we're registered for the whole thing